source/common/http/utility.cc

```diff
@@ -756,6 +757,62 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
   encodeLocalReply(is_reset, std::move(prepared_local_reply));
 }
 
+bool Utility::remoteAddressIsTrustedProxy(
+    const Envoy::Network::Address::InstanceConstSharedPtr& remote,
+    const std::vector<Network::Address::CidrRange> trusted_cidrs) {
+  for (const auto& cidr : trusted_cidrs) {
+    if (cidr.isInRange(*remote.get())) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
`const Envoy::Network::Address::Instance& remote`: const reference to shared pointer here make no sense.
`absl::Span<const Network::Address::CidrRange>`: to avoid copy whole vector. (or vector reference should be used.)

Reply:
Thank you, these have been updated.